### PR TITLE
🌱 added tests for version.go

### LINKED ISSUE #3211: Increase the test coverage with unit tests
### What do you want to happen?

By looking at the coveralls reports we can see that we have some implementations which could be better covered. We do not cover 100% of the code but would be great increase the coverage and ensure that we have no red and orange files/dirs, see: https://coveralls.io/github/kubernetes-sigs/kubebuilder?branch=master

Example: 

We could add more tests for cmd_helpers.go, completiton.go and version.go:

Screenshot 2023-02-08 at 14 17 13

### Extra Labels

_No response_

### PR DESCRIPTION
**Pull request by khareyash05**
added tests for `pkg/cli/version.go`

Fixes #3211

### CODE REVIEW

**Review by camilamacedo86 — APPROVED** [review]

File: pkg/cli/version_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+		BeforeEach(func() {
+			cmd = cli.newVersionCmd()
+			cmd.SetOutput(buf)
+		})
+
+		It("prints the version number", func() {
+			err := cmd.Execute()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(buf.String()).To(Equal("1.0.0\n"))
```
Comment (camilamacedo86) [inline pkg/cli/version_test.go:47]: @khareyash05 it is failing 

Screenshot 2023-04-22 at 19 41 27

You can either run it locally to fix it out.

**Comment by khareyash05** [comment]
@camilamacedo86 Kindly review

**Comment by camilamacedo86** [comment]
Hi @khareyash05,

See that the test is still not working. Note that you can test it locally either:

>   [FAILED] Expected success, but got an error:
>       <*errors.errorString | 0xc0003f8f20>: 
>       version string
>       {s: "version string"}
>   In [It] at: /home/prow/go/src/sigs.k8s.io/kubebuilder/pkg/cli/cli_test.go:477 @ 04/23/23 06:30:54.512 

We are more than happy to get this one merged when it be fixed.
Could you please look into that?

/approved cancel

**Pull request closed without merging**